test/test_other.py

```diff
@@ -12038,6 +12038,28 @@ def test_wasmfs_jsfile_proxying_backend(self):
     self.set_setting('EXIT_RUNTIME')
     self.test_wasmfs_jsfile()
 
+  @node_pthreads
+  def test_wasmfs_print_override_pthread(self):
+    create_file('main.js', '''
+      const Test = require('./test.js');
+
+      async function main() {
+        await Test({
+          // world -> earth
+          print: (text) => console.log(text.replace('world', 'earth'))
+        });
+      }
+      main();
```

Comment:
Does this needs need to be using MODULARIZE/EXPORT_NAME?  Could it not just use `onRuntimeInitialized` to run this code?

Reply:
Does this bug now also effect non-wasmfs?   Any not just write this test without WASMFS at all?

Reply:
Simplified the test with commit 2be3a5353222f5f27e2a835caf118013dd49fc51. Note that this bug only occurs when the handler is set outside the module.